bot/exts/backend/error_handler.py

```diff
@@ -15,6 +18,53 @@
 log = get_logger(__name__)
 
 
+class CheckedDeleteMessageButton(DeleteMessageButton):
+    """Button to delete a message, restricted to the message owner and moderators."""
+
+    def __init__(self, owner: User | Member):
+        super().__init__()
+        self.owner = owner
+
+    async def callback(self, interaction: Interaction) -> None:
+        """Delete the message if the user is the owner of the message or a moderator."""
+        user = interaction.user
+        if (
+            user.id == self.owner.id or
+            (
+                isinstance(user, Member) and
+                any(user.get_role(role) for role in MODERATION_ROLES)
+            )
+        ):
+            await super().callback(interaction)
+        else:
+            await interaction.response.send_message(
+                "You can only delete responses to your own command invocations!", ephemeral=True
+            )
+
+
+class HelpEmbedView(discord.ui.View):
```

Comment:
If you subclass [pydis_core.utils.interactions.ViewWithUserAndRoleCheck](https://bot-core.pythondiscord.com/main/output/pydis_core.utils.interactions.html#pydis_core.utils.interactions.ViewWithUserAndRoleCheck) instead, you can pass it allowed_users and allowed_roles for owner & mods, this will allow you to drop the subclass of DeleteMessageButton entirely and use it directly.

It will also let you remove the on_timeout handler, since you can set `.message` on the view and it will delete on timeout automatically.

Reply:
I think I tried to use that but couldn't find a way to limit the delete button to invoker & mods while still allowing anyone to use the help button.

Reply:
I mentioned that in an edit to me last message in the last paragraph, I'm not entirely sure that's needed.

If we completely set on needing that, then we could use that view from bot-core to handle the view itself, and just pass the `@everyone` role as the allowed roles, then the delete button can retain your callback code.

This then uses the timeout and stop code from bot core, which also covers things like if the message being editing is in a closed thread, which would error under this implementation

Reply:
I found a way to override the interaction check to let everyone use the help command https://github.com/python-discord/bot/pull/2439/commits/c053afe871543c79de241e137d2ab91b581b3ab4#diff-9a635f11cfc93b25d2c440dc75d5f492d682fd9770feebc4e5216000929b9ab9R30-R42, a little hacky but I think it's nicer than having to have the whole deletion button logic. Thoughts?